### PR TITLE
Add `default_to_square_for_size` to `CLIPImageProcessor`

### PR DESCRIPTION
**Pull request by ydshieh**
# What does this PR do?

Add `default_to_square_for_size` to `CLIPImageProcessor`.

As in the file, we also have `crop_size = get_size_dict(crop_size, default_to_square=True, param_name="crop_size")`, so I give the name `default_to_square_for_size` for `get_size_dict(size, default_to_square=default_to_square_for_size)` to avoid confusion.

Otherwise, I can give it `default_to_square`, and we may add `default_to_square_for_crop_size` later (or in this PR).

### CODE REVIEW

File: src/transformers/models/clip/image_processing_clip.py [hunk 1]
```diff
@@ -101,11 +103,12 @@ def __init__(
         image_mean: Optional[Union[float, List[float]]] = None,
         image_std: Optional[Union[float, List[float]]] = None,
         do_convert_rgb: bool = True,
+        default_to_square_for_size: bool = False,
```
Comment (ArthurZucker) [inline src/transformers/models/clip/image_processing_clip.py:106]:
```suggestion
        use_square_size: bool = False,
```
or something like this!

File: src/transformers/models/clip/image_processing_clip.py [hunk 2]
```diff
@@ -84,6 +84,8 @@ class CLIPImageProcessor(BaseImageProcessor):
             Can be overridden by the `image_std` parameter in the `preprocess` method.
         do_convert_rgb (`bool`, *optional*, defaults to `True`):
             Whether to convert the image to RGB.
+        default_to_square_for_size (`bool`, *optional*, defaults to `False`):
+            The value to be passed to `get_size_dict` as `default_to_square` when computing the image size.
```
Comment (ArthurZucker) [inline src/transformers/models/clip/image_processing_clip.py:88]: let's say what it changes depending on this values!

**Review by ArthurZucker — COMMENTED** [review]
few very small nits

File: src/transformers/models/clip/image_processing_clip.py (hunk 2 above)
Comment (ydshieh) [reply src/transformers/models/clip/image_processing_clip.py]:
```
        use_square_size (`bool`, *optional*, defaults to `False`):
            The value to be passed to `get_size_dict` as `default_to_square` when computing the image size. If the
            `size` argument in `get_size_dict` is an `int`, it determines whether to default to a square image or not.
            Note that this attribute is not used in computing `crop_size` via calling `get_size_dict`.
```

File: src/transformers/models/clip/image_processing_clip.py [hunk 3]
```diff
@@ -84,8 +84,10 @@ class CLIPImageProcessor(BaseImageProcessor):
             Can be overridden by the `image_std` parameter in the `preprocess` method.
         do_convert_rgb (`bool`, *optional*, defaults to `True`):
             Whether to convert the image to RGB.
-        default_to_square_for_size (`bool`, *optional*, defaults to `False`):
-            The value to be passed to `get_size_dict` as `default_to_square` when computing the image size.
+        use_square_size (`bool`, *optional*, defaults to `False`):
+            The value to be passed to `get_size_dict` as `default_to_square` when computing the image size. If the
+            `size` argument in `get_size_dict` is an `int`, it determines whether to default to a square image or not.
+            Note that this attribute is not used in computing `crop_size` via calling `get_size_dict`.
```
Comment (ArthurZucker) [inline src/transformers/models/clip/image_processing_clip.py:90]: Okay good enough for me, would be better to know what size are used if false for example

**Review by ArthurZucker — APPROVED** [review]
Thanks! I am pinging @amyeroberts so that she sees it when she comes back. Feel free to merge

**Comment by ydshieh** [comment]
Thanks a lot !

**Comment by ydshieh** [comment]
I forgot to change this line

```
        output_size = get_resize_output_image_size(
            image, size=size["shortest_edge"], default_to_square=False, input_data_format=input_data_format
        )
```
to 
```
        output_size = get_resize_output_image_size(
            image, size=size["shortest_edge"], default_to_square=self.use_square_size, input_data_format=input_data_format
        )
```
will do it but wait on next Monday to merge

**Review by amyeroberts — APPROVED** [review]
Thanks for adding!

**Pull request merged**